tasks/tasks.go

```diff
@@ -0,0 +1,93 @@
+package tasks
+
+import (
+	"errors"
+	"math/rand"
+	"sync"
+	"time"
+)
+
+// Status is the status of the task.
+type Status int
+
+// TaskExecutor executes a series of task functions.
+type TaskExecutor interface {
+	QueueTask(task func() error) string
+	GetTaskStatus(id string) (Status, error)
+	completeTask(id string, task func() error)
+}
+
+type defaultExecuter struct {
+	sync.RWMutex
+	m map[string]Status
```

Comment:
It does if you need to compare the status from another package as the `GetTaskStatus` returns the Status.